### PR TITLE
Feature/14515 answer service

### PR DESCRIPTION
**Pull request by GuyHarwood**
First version of Answer Service.

Small modification to Storage Service included to restrict keys used for storage items (see StorageKey class).

### CODE REVIEW

File: pupil-spa/package.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "lint": "ng lint",
     "e2e": "ng e2e",
     "copygovuktoolkit": "node tools/copyGovukFrontendToolkit",
-    "copygovukelements": "node tools/copyGovukElements"
+    "copygovukelements": "node tools/copyGovukElements",
+    "postinstall": "node tools/copyGovukFrontendToolkit && node tools/copyGovukElements"
```
Comment (jon-shipley) [inline pupil-spa/package.json]: thanks for this one

File: pupil-spa/src/app/answer.service.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      this.storageService.getItem('answers')
+        .then((answers) => {
+          if(!answers){
+            answers = "[]";
+          }
+          let existingAnswers = JSON.parse(answers);
+          existingAnswers.push(answer);
+          this.storageService.setItem('answers', JSON.stringify(existingAnswers));
+          resolve();
+        }).catch((err) => {
```
Comment (jon-shipley) [inline pupil-spa/src/app/answer.service.ts]: This is missing a potential rejected promise from the StorageService.

E.g. 
```
this.storageService.getItem('answers')
.then(
  (answers) => {},
  (err) => {} // <-- Missing
)
.catch(err) => {}
```

File: pupil-spa/src/app/storage.service.spec.ts [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-import { StorageService } from './storage.service';
+import { StorageService, StorageKey } from './storage.service';
 
 let service: StorageService;
 
 describe('StorageService', () => {
 
   const shouldNotExecute = () => {
-    expect(1).toBe(2);
+    expect('this code not').toBe('executed');
```
Comment (jon-shipley) [inline pupil-spa/src/app/storage.service.spec.ts]: If you are passing `done()` into the test you can exit the test with an error by calling
`done(new Error('Should not have been called'))`

File: pupil-spa/package.json (hunk 1 above)
Comment (GuyHarwood) [inline pupil-spa/package.json]: @jon-shipley code updated

**Pull request merged**